src/modify.js

```diff
@@ -66,12 +101,122 @@ function rewriteAllFields(schema, configCallback, context) {
   });
 }
 
+function pickFields(originalSchema, pickConfig) {
+  if (!pickConfig) return originalSchema;
+
+  const fieldsToPick = pickConfig.fields;
+  const newSchema = {
+    properties: {},
+  };
+
+  Object.entries(originalSchema).forEach(([attrKey, attrValue]) => {
+    switch (attrKey) {
+      case 'properties':
+        // TODO — handle recursive nested fieldsets
+        fieldsToPick.forEach((fieldPath) => {
+          set(newSchema.properties, fieldPath, attrValue[fieldPath]);
+        });
+        break;
+      case 'x-jsf-order':
+      case 'required':
+        newSchema[attrKey] = attrValue.filter((fieldName) => fieldsToPick.includes(fieldName));
+        break;
+      case 'allOf': {
+        const newAllOf = [];
+        // remove conditional ("if, then, else") if it does not contain any reference to fieldsToPick
+        originalSchema.allOf.forEach((condition) => {
+          if (isConditionalReferencingAnyPickedField(condition, fieldsToPick)) {
+            newAllOf.push(condition);
+          }
+        });
+        newSchema[attrKey] = newAllOf;
+
+        break;
+      }
+      default:
+        newSchema[attrKey] = attrValue;
+    }
+  });
+
+  // Look for unpicked fields in the conditionals...
+  let missingFields = {};
+
+  newSchema.allOf.forEach((condition, ix) => {
+    const { if: ifCondition, then: thenCondition, else: elseCondition } = condition;
+
+    missingFields = {
+      ...missingFields,
+      ...findMissingFields(ifCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].if`,
+      }),
+      ...findMissingFields(thenCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].then`,
+      }),
+      ...findMissingFields(elseCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].else`,
+      }),
+    };
+  });
+
+  if (Object.keys(missingFields).length > 0) {
+    // Read them to the new schema...
```

Comment:
I think this means `Add`

```suggestion
    // Add them to the new schema...
```